### PR TITLE
feat(breaking): UR address change for v4

### PR DESCRIPTION
**Pull request by dannydaniil**
## PR Scope

Please title your PR according to the following types and scopes following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/):

- `fix(SDK name):` will trigger a patch version
- `chore(<type>):` will not trigger any release and should be used for internal repo changes
- `<type>(public):` will trigger a patch version for non-code changes (e.g. README changes)
- `feat(SDK name):` will trigger a minor version
- `feat(breaking):` will trigger a major version for a breaking change

## Description

- This PR aims to introduce UR configs according to it's version
- It purposefully introduces a breaking change with the new required parameter so integrators are aware they're moving to a new router that requires new approvals etc etc
- Fixes API-437

## How Has This Been Tested?

_[e.g. Manually, E2E tests, unit tests, Storybook]_

## Are there any breaking changes?

Yes, `UNIVERSAL_ROUTER_ADDRESS` and `UNIVERSAL_ROUTER_CREATION_BLOCK`

If there are breaking changes, please ensure you bump the major version Bump the major version (by using the title `feat(breaking): ...`), post a notice in #eng-sdks, and explicitly notify all Uniswap Labs consumers of the SDK.

## (Optional) Feedback Focus

_[Specific parts of this PR you'd like feedback on, or that reviewers should pay closer attention to]_

## (Optional) Follow Ups

_[Things that weren't addressed in this PR, ways you plan to build on this work, or other ways this work could be extended]_

### CODE REVIEW

**Comment by ewilz** [comment]
Can we do something like:

```
export enum RouterVersion {
  V1_2,
  V2,
}

export const latestRouterVersion() {
    return RouterVersion.V2
}

export const UNIVERSAL_ROUTER_ADDRESS = (version: RouterVersion, chainId: number): string => {
....
} 
```

File: sdks/universal-router-sdk/src/utils/constants.ts [hunk 1]
```diff
@@ -1,8 +1,8 @@
 import { BigNumber } from 'ethers'
 
 type ChainConfig = {
-  v3Router: string
-  v4Router: string
+  v1_2Router: string
```
Comment (ewilz) [inline sdks/universal-router-sdk/src/utils/constants.ts:4]: we'll probably eventually have to do something similar with creationBlock. prefer alpha order based on thing, not version?? hence `routerV1_2`, `routerV2_0`? then we can have creationBlockV-whatever down the line

Comment (dannydaniil) [reply sdks/universal-router-sdk/src/utils/constants.ts:4]: great callout!

**Review by jsy1218 — APPROVED** [review]

File: sdks/universal-router-sdk/src/utils/constants.ts [hunk 2]
```diff
@@ -1,129 +1,285 @@
...
+      [RouterVersion.V2_0]: {
+        address: '0x28731BCC616B5f51dD52CF2e4dF0E78dD1136C06',
+        creationBlock: 12640979,
+      },
+    },
   },
 }
 
-export const UNIVERSAL_ROUTER_ADDRESS = (chainId: number): string => {
+// Todo: Change `CHAIN_CONFIGS` to pull the UR address with v4
```
Comment (ewilz) [inline sdks/universal-router-sdk/src/utils/constants.ts:273]: should the Todo on line 20 suffice? These seem noisy. I don't think this code will change here

Comment (dannydaniil) [reply sdks/universal-router-sdk/src/utils/constants.ts]: addressed!

**Review by ewilz — APPROVED** [review]

**Pull request merged**